R/plotting.R

```diff
@@ -510,72 +620,98 @@ plotMotifHeatmaps <- function(x,
+#' @param se the \code{SummarizedExperiment} object resulting from stability
 #'   selection, by running \code{\link[monaLisa]{randLassoStabSel}}.
 #' @param selProbMin A numerical scalar in [0,1]. Predictors with a selection
 #'   probability greater than \code{selProbMin} are shown as colored lines. The
 #'   color is defined by the \code{col} argument.
-#' @param col color of the selected predictors.
-#' @param lwd line width (default = 1).
-#' @param lty line type (default = 1).
-#' @param ylim limits for y-axis (default = c(0,1.1)).
-#' @param ... additional parameters to pass on to \code{matplot}.
-#'
-#' @return \code{TRUE} (invisibly).
-#' 
-#' @examples 
+#' @param col vector of colors specifying what to color selected and
+#'   non-selected predictors.
+#' @param linewidth line width.
+#' @param alpha line transparency of the stability paths.
+#' @param ylim limits for y-axis.
+#'
+#' @return a \code{ggplot2} object.
+#'
+#' @examples
 #' ## create data set
 #' Y <- rnorm(n = 500, mean = 2, sd = 1)
 #' X <- matrix(data = NA, nrow = length(Y), ncol = 50)
 #' for (i in seq_len(ncol(X))) {
 #'   X[ ,i] <- runif(n = 500, min = 0, max = 3)
 #' }
-#' s_cols <- sample(x = seq_len(ncol(X)), size = 10, 
+#' s_cols <- sample(x = seq_len(ncol(X)), size = 10,
 #'   replace = FALSE)
 #' for (i in seq_along(s_cols)) {
 #'   X[ ,s_cols[i]] <- X[ ,s_cols[i]] + Y
 #' }
-#'   
+#'
 #' ## reproducible randLassoStabSel() with 1 core
 #' set.seed(123)
 #' ss <- randLassoStabSel(x = X, y = Y)
 #' plotStabilityPaths(ss)
 #'
-#' @seealso \code{\link[stabs]{stabsel}} and \code{\link[graphics]{matplot}}
+#' @seealso \code{\link[stabs]{stabsel}}
 #'
 #' @importFrom SummarizedExperiment assay rowData colData
-#' @importFrom graphics matplot
+#' @import ggplot2
+#' @importFrom tidyr pivot_longer starts_with
+#' @importFrom cli cli_abort
 #'
 #' @export
 plotStabilityPaths <- function(se,
                                selProbMin = metadata(se)$stabsel.params.cutoff,
-                               col = "cadetblue", 
-                               lwd = 1, lty = 1, ylim = c(0, 1.1), ...) {
+                               col = c(sel = "cadetblue", notSel = "black"),
+                               linewidth = 0.5, alpha = 1, ylim = c(0, 1)) {
     # checks
     if (!is(se, "SummarizedExperiment")) {
-        stop("'se' must be a SummarizedExperiment")
+        cli_abort("{.arg se} must be a {.cls SummarizedExperiment}")
     }
+    .assertScalar(x = selProbMin, type = "numeric", rngIncl = c(0, 1))
+    .assertVector(x = col, type = "character", len = 2L)
+    .assertScalar(x = linewidth, type = "numeric")
+    .assertScalar(x = alpha, type = "numeric", rngIncl = c(0, 1))
+    .assertVector(x = ylim, type = "numeric", rngIncl = c(0, 1))
+    .assertVector(x = colnames(se), type = "character", len = ncol(se))
+    .assertVector(x = rownames(se), type = "character", len = nrow(se))
+    .assertVector(x = colnames(colData(se)), type = "character")
```

Comment:
Check already here that there are some columns starting with `regStep` (expected below)?

Reply:
Yeah I also wondered if I should add that. `regStep` should in principle be there, since it happens when calling `randLassoStabSel`, and the output of that should be fed into the plotting function. But can't hurt to have that extra check!